source/view.js

```diff
@@ -4119,12 +4133,38 @@ view.Tensor = class {
     }
 
     get metrics() {
-        const metrics = Array.from(this._tensor.metrics || []);
-        const keys = new Set(metrics.map((metrics) => metrics.name));
-        if (!keys.has('sparisity')) {
-            // metrics.push(new view.Argument('sparisity', 0, 'float32'));
+        if (this._metrics === null) {
+            const value = this.value;
```

Comment:
This leads to potentially large tensor values getting expanded immediately when the sidebar is opened. Would the metrics group need to be hidden initially via an expander or other UI?

Reply:
I assume that when you refer to "sidebar" you're referring to "tensor properties sidebar". However,  I've disabled the automatic expansion of tensor values in the tensor properties sidebar [here](https://github.com/lutzroeder/netron/pull/1293/files/5fbe0128f05b7f5638f46e3e5b657f326987c5c4#diff-93b7b9ad1ecc0b6d6d5ef6461d90bad02ca0b903792489204364e1d656cbf8eaL3165).

In terms of having the metrics section in the tensor properties sidebar being hidden by an expander, I think that could make sense in the future when more metrics are implemented.

Reply:
Not in the future, it will likely need to get addressed first. Accessing `this.value` can be an expensive operation and potentially lead to memory issues. For example, 500 MB tensor values in large models. 

Fundamental question is whether the Tensor sidebar should always expand content as the main value is showing content, render tensor visualization, compute metrics. In this is case the UX first needs to get hardened to indicate progress to the user and guard against errors. If not, all code paths accessing tensor content should be delayed by expanders.

Generally, the concern is less enabling metrics, more that it becomes a forcing function to prioritize the work to design and implement the Tensor sidebar, setting expectations with users that might need to get rolled back. There was no Tensor sidebar until now and the implementation is far from complete. For example, in many formats there is no away to activate the Tensor sidebar as there are no tensor names. Some formats have two layers making quantization part of tensor implementation. If usage increases these issues will need to get addressed or the feature disabled.

Reply:
In TensorFlow Lite for example, `Tensor` will contain the tensor data but also information to de-quantize. In other formats like Arm NN this happens in the `Value` wrapping an initializer. Tensor sidebar might need to evolve a set of features to support both a quantized and de-quantized view of the tensor data. The challenge is whether this can be generalized across the many different tensor formats and quantization approaches.